src/connector/src/sink/iceberg.rs

```diff
@@ -166,38 +199,113 @@ impl Sink for IcebergSink {
-pub struct IcebergWriter {}
+pub struct IcebergWriter {
+    is_append_only: bool,
+    table: Table,
+    writer: icelake::io::task_writer::TaskWriter,
+}
+
+impl IcebergWriter {
+    async fn append_only_write(&mut self, chunk: StreamChunk) -> Result<()> {
+        let (mut chunk, ops) = chunk.into_parts();
+
+        let filters =
+            Bitmap::from_bool_slice(&ops.iter().map(|op| *op == Op::Insert).collect_vec());
+        let filters = if let Some(ori_vis) = chunk.visibility() {
+            ori_vis & &filters
+        } else {
+            filters
+        };
+
+        chunk.set_visibility(filters);
+        let chunk = RecordBatch::try_from(&chunk.compact())
+            .map_err(|err| SinkError::Iceberg(err.to_string()))?;
+
+        self.writer.write(&chunk).await.map_err(|err| {
+            SinkError::Iceberg(format!("Write chunk fail: {}, chunk: {:?}", err, chunk))
+        })?;
+
+        Ok(())
+    }
+}
 
 #[async_trait]
 impl SinkWriter for IcebergWriter {
     /// Begin a new epoch
     async fn begin_epoch(&mut self, _epoch: u64) -> Result<()> {
-        todo!()
+        // Just skip it.
+        Ok(())
     }
 
     /// Write a stream chunk to sink
-    async fn write_batch(&mut self, _chunk: StreamChunk) -> Result<()> {
-        todo!()
+    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
+        if self.is_append_only {
+            self.append_only_write(chunk).await
+        } else {
+            return Err(SinkError::Iceberg(
+                "Iceberg sink only support append-only mode now.".to_string(),
+            ));
+        }
     }
 
     /// Receive a barrier and mark the end of current epoch. When `is_checkpoint` is true, the sink
     /// writer should commit the current epoch.
-    async fn barrier(&mut self, _is_checkpoint: bool) -> Result<()> {
-        todo!()
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<()> {
+        // Skip it if not checkpoint
+        if !is_checkpoint {
+            return Ok(());
+        }
+
+        let old_writer = std::mem::replace(
+            &mut self.writer,
+            self.table
+                .task_writer()
+                .await
+                .map_err(|err| SinkError::Iceberg(err.to_string()))?,
+        );
+
+        let mut tx = Transaction::new(&mut self.table);
+
+        let data_files = old_writer
+            .close()
+            .await
+            .map_err(|err| SinkError::Iceberg(format!("Close writer fail: {}", err)))?;
+        if data_files.is_empty() {
+            return Ok(());
+        }
+
+        tx.append_file(data_files.into_iter());
+        tx.commit()
+            .await
+            .map_err(|err| SinkError::Iceberg(err.to_string()))?;
+
+        Ok(())
     }
 
     /// Clean up
     async fn abort(&mut self) -> Result<()> {
-        todo!()
+        return Err(SinkError::Iceberg(
+            "Iceberg sink can't support abort now".to_string(),
```

Comment:
In `abort` we should delete finished but uncommitted files. We can postpone this in later pr, but we should not return the error here. We should just ignore it.